### ISSUE TITLE
Accessible status badges

### ISSUE THREAD

**Issue by gfellerph** [swisspost/design-system#5991]
### ✨ Is your feature request related to a problem? Please describe it.

With colorblindness, it's difficult to differentiate status badges by color. They are small and the status is just indicated by color alone. That violates (WCAG 1.4.1 use of color)[https://www.w3.org/WAI/WCAG21/Understanding/use-of-color.html]. 

### 💡 Describe the solution you'd like

In Teams, status badges come with small icons embedded that enhance the status information with a shape.

<img width="364" height="412" alt="Image" src="https://github.com/user-attachments/assets/fe8818d6-b26e-474c-909e-9ceab9f0f2af" />

### Tasks
- [ ] Update design and specify icons for the most common states error, success, warning, waiting, (and what else?)
- [ ] Update storybook documentation to show examples of status badges, currently only number badges are shown

**Comment by gfellerph** [comment]
Get usage examples and check if it's covered by https://www.figma.com/design/JIT5AdGYqv6bDRpfBPV8XR/Foundations---Components-Next-Level?t=zLky2Oe5lLoB3ITX-0

Usage sample: Badge on an avatar for availability status

<img width="1600" height="800" alt="Image" src="https://github.com/user-attachments/assets/8e80927d-0632-4288-aa1d-7ddbcfd86984" />

**Comment by gfellerph** [comment]
Pattern discussion: Should we add the user status **pattern**, consisting of avatar, badge and icon, to the documentation in figma and storybook? This would not be a standalone component, but a pattern cobbled together from 3 different components.

**Comment by leagrdv** [comment]
@sandra-post to discuss with Rouven

**Comment by alizedebray** [comment]
If we decide to implement these statuses, will we use current badges with an icon inside or rather the signal icons?

**Comment by rouvenpost** [comment]
I created the Icons: https://www.figma.com/design/q57uYXNspU7G1pYEnDGTyM/Icons-Next-Level?node-id=5211-6891&t=fNkKGbEc3Pnbk0LR-1

**Comment by rouvenpost** [comment]
[Available.zip](https://github.com/user-attachments/files/22487742/Available.zip)
[BeRightBack.zip](https://github.com/user-attachments/files/22487740/BeRightBack.zip)
[Busy.zip](https://github.com/user-attachments/files/22487741/Busy.zip)
[DoNotDisturb.zip](https://github.com/user-attachments/files/22487744/DoNotDisturb.zip)
[Offline.zip](https://github.com/user-attachments/files/22487743/Offline.zip)

**Comment by rouvenpost** [comment]
[Appear Away.zip](https://github.com/user-attachments/files/22487769/Appear.Away.zip)

**Comment by rouvenpost** [comment]
The Icons in Figma can be found here: https://www.figma.com/design/q57uYXNspU7G1pYEnDGTyM/Icons-Next-Level?node-id=5263-6595&t=vT1V47EvD6WUPKq8-1

**Comment by gfellerph** [comment]
Icons are available in the library. Generally, the need for a dedicated component is not high enough to meet the threshold for adding a new component to the system. I'm closing this ticket for the moment. If we see that there are more use-cases, we can re-open the discussion.
